hibernate-core/src/main/java/org/hibernate/engine/jdbc/connections/spi/AbstractMultiTenantConnectionProvider.java

```diff
@@ -48,15 +48,17 @@ public boolean supportsAggressiveRelease() {
 
 	@Override
 	public boolean isUnwrappableAs(Class unwrapType) {
-		return ConnectionProvider.class.equals( unwrapType ) ||
-				MultiTenantConnectionProvider.class.equals( unwrapType ) ||
+		return MultiTenantConnectionProvider.class.equals( unwrapType ) ||
 				AbstractMultiTenantConnectionProvider.class.isAssignableFrom( unwrapType );
 	}
 
 	@Override
-	@SuppressWarnings( {"unchecked"})
+	@SuppressWarnings({"unchecked"})
 	public <T> T unwrap(Class<T> unwrapType) {
-		if ( isUnwrappableAs( unwrapType ) ) {
+		if ( ConnectionProvider.class.equals( unwrapType ) ) {
```

Comment:
The new `AbstractMultiTenantConnectionProvider#isUnwrappableAs` contradicts the ability to unwrap to a `ConnectionProvider`.